client/lib/domains/dns/index.js

```diff
@@ -45,8 +46,8 @@ function isValidDomain( name ) {
 	return /^([a-z0-9-_]{1,63}\.)*[a-z0-9-]{1,63}\.[a-z]{2,63}$/i.test( name );
 }
 
-function isValidName( name, type ) {
-	if ( isNameEmptyAndItsAllowed( name, type ) ) {
+function isValidName( name, type, domainName ) {
+	if ( isRootDomainAndItsAllowed( name, type, domainName ) ) {
```

Comment:
`andIsValid` makes more sense, but not entirely. I don't want to stall this PR but it doesn't make sense with "allowed". What does it mean to be allowed? What's the rule behind being allowed? In this case, it seems there are certain types of records that can be root domain. So why not `isRootDomain(name, domainName) && canBeRootDomain(type)`? Or create constant for `ALLOWED_ROOT_DOMAIN_TYPES` and do the check inline?